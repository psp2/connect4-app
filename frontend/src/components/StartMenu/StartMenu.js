import React, { useState } from "react";
import "../../pages/Menus/Menus.scss";

import MenuInfo from "../MenuInfo/MenuInfo";

function StartMenu(props) {
  const [infoDisplay, setInfoDisplay] = useState(null);

  function selectNumPlayers(num) {
    props.changeNumberPlayers(num);
  }

  function createInfoDisplay(displayType) {
    setInfoDisplay(displayType);
  }

  function removeInfoDisplay() {
    setInfoDisplay(null);
  }

  return (
    <div>
      <div className="MenuBoard">
        <div className="MenuTitle">
          <div className="MenuTitleWhite">Connect</div>
          <div className="MenuTitleRed">4</div>
        </div>
        <div className="MenuRow">
          <div
            className="WhiteButton"
            onClick={() => createInfoDisplay("tutorial")}
          >
            <div className="ButtonCaption">How to Play</div>
          </div>
          <div className="YellowButton" onClick={() => selectNumPlayers(1)}>
            <div className="ButtonCaption">Single Player</div>
          </div>
          <div className="GreenButton" onClick={() => selectNumPlayers(2)}>
            <div className="ButtonCaption">Multi-Player</div>
          </div>
          <div className="RedButton" onClick={() => createInfoDisplay("about")}>
            <div className="ButtonCaption">About the Game</div>
          </div>
        </div>
      </div>
      <div>
        {infoDisplay && (
          <MenuInfo
            displayType={infoDisplay}
            closeDisplay={removeInfoDisplay}
          />
        )}
      </div>
    </div>
  );
}

export default StartMenu;
