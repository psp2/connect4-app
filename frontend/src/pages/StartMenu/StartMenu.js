import React, { useState } from "react";
import "./StartMenu.scss";
import { compose } from "recompose";
import { withRouter, Redirect } from "react-router-dom";
import { Button } from "react-bootstrap";

function StartMenu() {
  const [gameRedirect, setGameRedirect] = useState(false);

  if (gameRedirect) {
    return <Redirect to="/game" />;
  }

  return (
    <div>
      This is the start menu (to be implemented in Week 2).
      <Button onClick={() => setGameRedirect(true)}>CLICK FOR GAME PAGE</Button>
    </div>
  );
}

export default compose(withRouter)(StartMenu);
