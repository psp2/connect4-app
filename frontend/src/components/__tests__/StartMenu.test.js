import React from "react";
import StartMenu from "../StartMenu/StartMenu";
import renderer from "react-test-renderer";

it("Start Menu, renders correctly", () => {
  const tree = renderer.create(<StartMenu />).toJSON();
  expect(tree).toMatchSnapshot();
});
