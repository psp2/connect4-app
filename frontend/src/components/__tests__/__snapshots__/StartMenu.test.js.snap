// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start Menu, renders correctly 1`] = `
<div>
  <div
    className="MenuBoard"
  >
    <div
      className="MenuTitle"
    >
      <div
        className="MenuTitleWhite"
      >
        Connect
      </div>
      <div
        className="MenuTitleRed"
      >
        4
      </div>
    </div>
    <div
      className="MenuRow"
    >
      <div
        className="WhiteButton"
        onClick={[Function]}
      >
        <div
          className="ButtonCaption"
        >
          How to Play
        </div>
      </div>
      <div
        className="YellowButton"
        onClick={[Function]}
      >
        <div
          className="ButtonCaption"
        >
          Single Player
        </div>
      </div>
      <div
        className="GreenButton"
        onClick={[Function]}
      >
        <div
          className="ButtonCaption"
        >
          Multi-Player
        </div>
      </div>
      <div
        className="RedButton"
        onClick={[Function]}
      >
        <div
          className="ButtonCaption"
        >
          About the Game
        </div>
      </div>
    </div>
  </div>
  <div />
</div>
`;
